src/Symfony/Component/AssetMapper/ImportMap/ImportMapRenderer.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\AssetMapper\ImportMap;
+
+/**
+ * @experimental
+ *
+ * @author Kévin Dunglas <kevin@dunglas.dev>
+ * @author Ryan Weaver <ryan@symfonycasts.com>
+ */
+class ImportMapRenderer
+{
+    public function __construct(
+        private readonly ImportMapManager $importMapManager,
+        private readonly string $charset = 'UTF-8',
+        private readonly string|false $polyfillUrl = ImportMapManager::POLYFILL_URL,
+        private readonly array $scriptAttributes = [],
+    ) {
+    }
+
+    public function render(?string $entryPoint = null): string
+    {
+        $attributeString = '';
+        foreach ($this->scriptAttributes as $name => $value) {
+            $attributeString .= ' ';
+            if ($value === true) {
+                $attributeString .= $name;
+
+                continue;
+            }
+            $attributeString .= sprintf('%s="%s"', $name, $this->escapeAttributeValue($value));
+        }
+
+        $output = <<<HTML
+            <script type="importmap"{$attributeString}>
+            {$this->importMapManager->getImportMapJson()}
+            </script>
+            HTML;
+
+        if ($this->polyfillUrl) {
+            $url = $this->escapeAttributeValue($this->polyfillUrl);
+
+            $output .= <<<HTML
+
+                <!-- ES Module Shims: Import maps polyfill for modules browsers without import maps support -->
+                <script async src="$url"$attributeString></script>
+                HTML;
+        }
+
+        foreach ($this->importMapManager->getModulesToPreload() as $url) {
+            $url = htmlspecialchars($url, \ENT_COMPAT | \ENT_SUBSTITUTE, $this->charset);
+
+            $output .= "\n<link rel=\"modulepreload\" href=\"{$url}\">";
+        }
+
+        if (null !== $entryPoint) {
+            $output .= "\n<script type=\"module\">import '$entryPoint';</script>";
```

Comment:
do we need some escaping of `$entryPoint`?

Reply:
We definitely do need escaping in a JS context

Reply:
@stof can you mention a problematic input? Obviously if `$entrypoint` had a `'` in it, that'd be a problem. But that's not allowed. What are the cases we need to escape? thx :)

Reply:
The class does not make any validation about the content of `$entrypoint`. So anything is allowed.

Reply:
> Obviously if $entrypoint had a ' in it, that'd be a problem. But that's not allowed

Actually, I see that I was wrong about this - `'` would be possible to allow. So we should escape that. Anything else? I don't think we need to escape anything else because this argument is coming from the developer. Also, if we escape `'`, then... would it even be theoretically possible for someone to close the string and inject XSS? I don't think so...